### ISSUE TITLE
List processing incorrectly marked with W8402 (use-list-copy)

### ISSUE THREAD

**Issue by Huy-Ngo** [tonybaloney/perflint#58]
When I use `dict.setdefault` to create a list and append to it to sort a list into several lists, perflint mistakes this as a list copy:

```python
r = range(10)
a = {}

for i in r:  # W8402: Use a list copy instead of a for-loop (use-list-copy)
    a.setdefault(i % 2, []).append(i)
```
